starknet_devnet/__init__.py

```diff
@@ -18,34 +18,43 @@
 
 def _patch_pedersen_hash():
     """
-    This is a monkey-patch to improve the performance of the devnet
-    We are using c++ code for calculating the pedersen hashes
-    instead of python implementation from cairo-lang package
+    Improves performance by substituting the default Python implementation of Pedersen hash
+    with swm's Python wrapper of C++ implementation.
     """
 
-    import starkware.cairo.lang.vm.crypto
-    from crypto_cpp_py.cpp_bindings import cpp_hash
-    from starkware.crypto.signature.fast_pedersen_hash import pedersen_hash
+    import starkware.crypto.signature.fast_pedersen_hash
+    from crypto_cpp_py.cpp_bindings import cpp_hash as patched_pedersen_hash
 
-    def patched_pedersen_hash(left: int, right: int) -> int:
-        """
-        Pedersen hash function written in c++
-        """
-        return cpp_hash(left, right)
+    starkware.crypto.signature.fast_pedersen_hash.pedersen_hash = patched_pedersen_hash
 
-    setattr(
-        sys.modules["starkware.crypto.signature.fast_pedersen_hash"],
-        "pedersen_hash",
-        patched_pedersen_hash,
+
+_patch_pedersen_hash()
+
+
+def _patch_poseidon_hash():
+    """
+    Improves performance by substituting the default Python implementation of Poseidon hash
+    with swm's Python wrapper of C implementation.
```

Comment:
same thing

Reply:
Addressed in https://github.com/0xSpaceShard/starknet-devnet/pull/452/commits/8e7f066d68ff6f2de641ffd0bff524a44f6aee8d